test/ccall.jl

```diff
@@ -123,7 +126,11 @@ function test_struct1{Struct}(::Type{Struct})
     b = Float32(123.456)
 
     a2 = copy(a)
-    x = ccall((:test_1, libccalltest), Struct, (Struct, Float32), a2, b)
+    if Struct === Struct1
```

Comment:
The old version was taking advantage of the fact that we would usually compile a new version of the method for every input type. This PR removes that constraint from codegen so we can make it more robust.